tardis/tests/tests_slow/report.py

```diff
@@ -0,0 +1,233 @@
+"""
+A helper class which works as a plugin to generate the test report and upload it
+to the group server's dokuwiki. It writes test execution details in dokuwiki
+syntax strings. The test report contains the following details:
+
+* The git commit hash on which test run was executed.
+* The time of generation of test report.
+* Number of passes, fails, errors, skips etc.
+* Tabular representation of each method - name, result, duration.
+* Embedded image of plot(s) and error log below a particular method (if any).
+
+When the integration tests are selected for a particular test run, this class
+is registered as a plugin in `pytest_configure` and subsequently unregistered in
+`pytest_unconfigure`. As a plugin, it implements several "hook" functions
+specified in pytest's official documentation.
+
+References
+==========
+1. "Writing Plugins" ( https://pytest.org/latest/writing_plugins.html )
+2. "Hookspec Source" ( https://pytest.org/latest/_modules/_pytest/hookspec.html )
+3. "pytest-html" ( https://www.github.com/davehunt/pytest-html )
+"""
+
+from datetime import datetime
+import tempfile
+import time
+
+from pytest_html.plugin import HTMLReport
+from astropy.tests.helper import remote_data
+import tardis
+
+# For specifying error while exception handling
+from socket import gaierror
+
+try:
+    import dokuwiki
+except ImportError:
+    dokuwiki_available = False
+else:
+    dokuwiki_available = True
+
+
+@remote_data
+class DokuReport(HTMLReport):
+
+    def __init__(self, dokuwiki_details):
+        """
+        Initialization of a DokuReport object and registration as a plugin
+        occurs in `pytest_configure`, where a dict containing url, username and
+        password of dokuwiki is passed through `dokuwiki_details`.
+        """
+        logfile = tempfile.TemporaryFile()
+        super(DokuReport, self).__init__(logfile.name)
+        self.logfile = logfile
+
+        # This list stores tuples of name, result, duration of each test in
+        # dokuwiki syntax strings.
+        self.test_logs = []
+        # This list stores strings containing failure log (if any)
+        # If not, empty strings are appended.
+        self.error_logs = []
+        # This list contains dokuwiki formatted strings of links to comparison
+        # plot images (if any). If not, empty strings are appended.
+        self.plot_images = []
+        # Length of all three lists are same.
+
+        # `generated` stores the date and time of generation of report.
+        # It is set in `pytest_sessionfinish` using datetime.now()
+        self.generated = None
+
+        # File path of image of comparison plots for a single test.
+        self.image_path = None
+
+        # Make a dokuwiki connection if dokuwiki import was successful.
+        if dokuwiki_available:
```

Comment:
you don't need this.
In the try/except for importing just define `dokuwiki = None` on import error
then you can catch a TypeError? in the next line.